editoast/src/views/timetable/paced_train.rs

```
@@ -681,6 +685,143 @@ async fn project_path(
+        operational_points_distances,
+    }): Json<ProjectPathOperationalPointForm>,
+) -> Result<Json<HashMap<i64, ProjectPathPacedTrainResult>>> {
+    let infra = &Infra::retrieve_real_or_fail(db_pool.get().await?, infra_id, || {
+        PacedTrainError::InfraNotFound { infra_id }
+    })
+    .await?;
+
+    let authorized = auth
+        .check_roles([authz::Role::OperationalStudies, authz::Role::Stdcm].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    auth.check_authorization(async |authorizer| {
+        authorizer
+            .authorize_infra_read(&authz::Infra(infra_id))
+            .await
+    })
+    .await?;
+
+    let conn = &mut db_pool.get().await?;
+
+    let paced_trains: Vec<models::PacedTrain> =
+        models::PacedTrain::retrieve_batch_or_fail(conn, train_ids, |missing| {
+            PacedTrainError::BatchNotFound {
+                count: missing.len(),
+            }
+        })
+        .await?;
+
+    let (ids, train_schedules): (Vec<_>, Vec<_>) = paced_trains
+        .iter()
+        .flat_map(|paced_train| {
+            std::iter::once((
+                ExceptionOrPacedTrain::PacedTrain {
+                    paced_train_id: paced_train.id,
+                },
+                paced_train.clone().into_train_schedule(),
+            ))
+            .chain(paced_train.exceptions.iter().map(|exception| {
+                (
+                    ExceptionOrPacedTrain::Exception {
+                        paced_train_id: paced_train.id,
+                        exception_key: exception.key.clone(),
+                    },
+                    paced_train.apply_exception(exception),
+                )
+            }))
+        })
+        .collect();
+
+    let operational_points_projection = OperationalPointProjection {
+        ids: operational_points_ids,
+        distances: operational_points_distances,
+    };
```

This can be transformed into a constructor that checks if `ids.len() == distances.len() + 1` and `ids.len() >= 2`.
If not, we can return an error that is forwarded to the frontend.